celery/contrib/sphinx.py

```diff
@@ -29,7 +29,23 @@
 syntax.
 
 Use ``.. autotask::`` to alternatively manually document a task.
+
+Sphinx 9.0+ Compatibility
```

Comment:
https://github.com/sphinx-doc/sphinx/releases/tag/v9.0.0 was released while the CI was running and broke some jobs. This enables the backward compatibility flag to revert to the previous behavior: [autodoc_use_legacy_class_based](https://www.sphinx-doc.org/en/master/usage/extensions/autodoc.html#confval-autodoc_use_legacy_class_based)